src/lightmap.cpp

```diff
@@ -651,6 +652,14 @@ float map::light_transparency( const tripoint &p ) const
     return get_cache_ref( p.z ).transparency_cache[p.x][p.y];
 }
 
+void map::get_lightmap( const int zlev )
+{
+    if( level_cache::lightmap_latest_zlevel == zlev ) {
```

Comment:
This will only prevent spurious recaching if the NPCs are processed by z-level and only once per turn.